### PR TITLE
Update SDK reference documentation for AutoML

### PR DESCRIPTION
**Pull request by harneetvirk**
# Description

Update SDK reference documentation for AutoML

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/automl_job.py [hunk 1]
```diff
@@ -34,7 +34,28 @@
 
 
 class AutoMLJob(Job, JobIOMixin, AutoMLNodeIOMixin, ABC):
-    """AutoML job entity."""
+    """Initialize an AutoML job entity.
+
+    Constructor for an AutoMLJob.
```
Comment (raduk) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/automl_job.py:39]: why replicate the comment from below here?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/automl_job.py [hunk 2]
```diff
@@ -48,13 +69,25 @@ def __init__(
...
-        :type queue_settings: QueueSettings
-        :param kwargs:
+        Constructor for an AutoMLJob.
+
+        :keyword resources: Resource configuration for the AutoML job, defaults to None
+        :paramtype resources: typing.Optional[ResourceConfiguration]
+        :keyword identity: Identity that training job will use while running on compute, defaults to None
+        :paramtype identity: typing.Optional[ typing.Union[ManagedIdentityConfiguration, AmlTokenConfiguration
+            , UserIdentityConfiguration] ]
+        :keyword environment_id: The environment id for the AutoML job, defaults to None
```
Comment (raduk) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/automl_job.py:79]: these 3 are not parameters of init ..

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/automl_vertical.py [hunk 3]
```diff
@@ -11,12 +11,48 @@
 
 
 class AutoMLVertical(AutoMLJob):
-    """Abstract class for AutoML verticals."""
+    """Abstract class for AutoML verticals.
+
+    :param task_type: The type of task to run. Possible values include: "classification", "regression", "forecasting".
```
Comment (raduk) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/automl_vertical.py:16]: same as above for the replication

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/automl_job.py (hunk 1 above)
Comment (harneetvirk) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/automl_job.py:39]: Both the class and constructor must be documented.
__init__ docstring is not visible on the learning docs portal, but the docstring from init method is extracted on running `help(YourClassName. __init__)`. A good strategy to follow is to make your changes in the class docstring and then copy the changes to the __init__ method docstring.

**Review by raduk — CHANGES_REQUESTED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/automl_job.py (hunk 2 above)
Comment (harneetvirk) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/automl_job.py:79]: Those 3 parameters are being retrieve from the kwargs. Here is the code copied from the init method: 

`

        kwargs[TYPE] = JobType.AUTOML
        self.environment_id = kwargs.pop("environment_id", None)
        self.environment_variables = kwargs.pop("environment_variables", None)
        self.outputs = kwargs.pop("outputs", None)

        super().__init__(**kwargs)
`

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/automl_vertical.py (hunk 3 above)
Comment (harneetvirk) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/automl_vertical.py:16]: Both the class and constructor must be documented.
__init__ docstring is not visible on the learning docs portal, but the docstring from init method is extract on running `help(YourClassName. __init__)`. A good strategy to follow is to make your changes in the class docstring and then copy the changes to the __init__ method docstring.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/automl_job.py (hunk 2 above)
Comment (raduk) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/automl_job.py:79]: in this case what is the drawback if we explicitly add them?

**Review by raduk — APPROVED** [review]

**Comment by harneetvirk** [comment]
Screenshot from the updated docs:

image


image



image

File: sdk/ml/azure-ai-ml/azure/ai/ml/constants/_common.py [hunk 4]
```diff
@@ -177,29 +177,53 @@
 )
 
 
-class AzureMLResourceType(object):
+class AzureMLResourceType:
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/constants/_common.py:180]: Is this in our public doc? This is a private namespace.

File: sdk/ml/azure-ai-ml/azure/ai/ml/constants/_compute.py [hunk 5]
```diff
@@ -3,78 +3,134 @@
 # ---------------------------------------------------------
 
 
-class ComputeType(object):
+class ComputeType:
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/constants/_compute.py:6]: Same question here. I thought these files are private interface.

File: sdk/ml/azure-ai-ml/azure/ai/ml/constants/_common.py (hunk 4 above)
Comment (harneetvirk) [reply sdk/ml/azure-ai-ml/azure/ai/ml/constants/_common.py:180]: All the objects(private/public) should be documented. We will be adding the gates to check for the docstring for classes/modules/methods, irrespective of public or private accessibility. This PR will serve as a sample PR to partner teams.

File: sdk/ml/azure-ai-ml/azure/ai/ml/constants/_compute.py (hunk 5 above)
Comment (harneetvirk) [reply sdk/ml/azure-ai-ml/azure/ai/ml/constants/_compute.py:6]: All the objects(private/public) should be documented.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/automl_job.py [hunk 6]
```diff
@@ -34,7 +34,28 @@
 
 
 class AutoMLJob(Job, JobIOMixin, AutoMLNodeIOMixin, ABC):
-    """AutoML job entity."""
+    """Initialize an AutoML job entity.
```
[the remark this answers is not in the document]
Comment (harneetvirk) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/automl_job.py:37]: The learning portal renders the docstrings mentioned in the class description. Saying that, we need to identify the properties used in the derived class and document them. 
Here is a link to the learning portal not showing all the attributes from the AutoML job entity: https://review.learn.microsoft.com/python/api/azure-ai-ml/azure.ai.ml.automl.classificationjob?view=azure-ml-py&branch=sdk-cli-v2-preview-master

**Pull request merged**